src/Kopernicus.Components/KopernicusStar.cs

```diff
@@ -155,10 +155,10 @@ public static Double Flux(ModularFlightIntegrator fi, KopernicusStar star)
                 // Get sunVector
                 RaycastHit raycastHit;
                 Boolean directSunlight = false;
-                Vector3d scaledSpace = ScaledSpace.LocalToScaledSpace(fi.IntegratorTransform.position);
+                Vector3d scaledSpace = ScaledSpace.LocalToScaledSpace(fi.Vessel.CoMD);
                 Double scale = Math.Max((star.sun.scaledBody.transform.position - scaledSpace).magnitude, 1);
-                Vector3 sunVector = (star.sun.scaledBody.transform.position - scaledSpace) / scale;
-                Ray ray = new Ray(ScaledSpace.LocalToScaledSpace(fi.IntegratorTransform.position), sunVector);
+                Vector3d sunVector = (star.sun.position - scaledSpace) / scale;
```

Comment:
This change would have serious implications on the calculation, because you are replacing a scaled space position with a local space one (scaled space = local space / 6000).